pkg/splunk/enterprise/util.go

```diff
@@ -47,11 +49,34 @@ func ApplySplunkConfig(client splcommon.ControllerClient, cr splcommon.MetaObjec
 		if err = splctrl.ApplyConfigMap(client, defaultsMap); err != nil {
 			return nil, err
 		}
+		// We will update the annotation for resource version in the pod template spec
+		// of the CR, so that any change in the ConfigMap will lead to recycle of the pod.
+		err = updateDefaultConfigMapAnnotations(client, cr, defaultsMap)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	return namespaceScopedSecret, nil
 }
 
+// updateDefaultConfigMapAnnotations checks and gets the default config map and updates the annotation
+// for the resource version of the ConfigMap in the pod template spec for the CR.
+func updateDefaultConfigMapAnnotations(client splcommon.ControllerClient, cr splcommon.MetaObject, configMap *corev1.ConfigMap) error {
+	namespacedName := types.NamespacedName{Namespace: configMap.GetNamespace(), Name: configMap.GetName()}
+	var current corev1.ConfigMap
+
+	err := client.Get(context.TODO(), namespacedName, &current)
+	if err != nil {
+		return err
+	}
+	annotationsMap := cr.GetObjectMeta().GetAnnotations()
```

Comment:
I am not sure if we have a reason to apply at CR object level. 
Also, I think the better place to add this is as part of updateSplunkPodTemplateWithConfig(). That way, the mount, and the annotation go together if we want to selectively apply the defaults configMap now(i see the Monitoring console not using it, at least for now) OR in the future.

Reply:
hmm not sure if the place would have mattered much since anyways we are appending the labels and annotations from the parent CR to the statefulset we would create or update. See `splcommon.AppendParentMeta(statefulSet.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())` in `getSplunkStatefulSet`